juno_magic/extensions/wamp.py

```diff
@@ -387,6 +391,46 @@ def execute(self, cell, prefix=None, **kwargs):
             output = yield self._wamp.call(".".join([prefix, "execute"]), cell)
         output = yield self._wamp.call(".".join([self._kernel_prefix, "execute"]), cell)
 
+    @inlineCallbacks
+    def manage_connections(self):
+        @inlineCallbacks
+        def ping():
+            # returns True or False if we are still connected 
+            # if True, it means everything is ok
+            # if False, it means the remote kernel client has died/is not active
+            res = yield self._wamp.call(".".join([self._wamp_prefix, u"ping"]))
+            returnValue(res)      
+        
+        @inlineCallbacks
+        def prefix_heartbeat():
+            if self._kernel_prefix is not None:
+                try:
+                    res = yield ping()
+                    if not res:
+                        print("Remote client currently inactive, try reconnecting later")
```

Comment:
Where do these print statements output, how does that relate to the twisted logging?